### PR TITLE
IGNITE-10071 Queued and running builds hang in the TC bot

### PR DESCRIPTION
**Pull request by dspavlov**
Fix for
Platform .NET [ tests 1 ]  72 queued

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -41,6 +41,7 @@ allprojects {
     ext {
         jettyVer = '9.4.12.v20180830'
 
+        //ignVer = '2.6.0'
```
Comment (SomeFire) [inline build.gradle]: Do we need it?

File: ignite-tc-helper-web/src/main/java/org/apache/ignite/ci/teamcity/ignited/fatbuild/ProactiveFatBuildSync.java [hunk 2]
```diff
@@ -124,28 +130,34 @@ public synchronized SyncTask getSyncTask(ITeamcityConn conn) {
     protected String findMissingBuildsFromBuildRef(String srvId, ITeamcityConn conn) {
         int srvIdMaskHigh = ITeamcityIgnited.serverIdToInt(srvId);
 
-        final int[] buildRefKeys = buildRefDao.getAllIds(srvIdMaskHigh);
+        Stream<BuildRefCompacted> buildRefs = buildRefDao.compactedBuildsForServer(srvIdMaskHigh);
 
         List<Integer> buildsIdsToLoad = new ArrayList<>();
-        int totalAskedToLoad = 0;
+        AtomicInteger totalAskedToLoad = new AtomicInteger();
```
Comment (SomeFire) [inline ignite-tc-helper-web/src/main/java/org/apache/ignite/ci/teamcity/ignited/fatbuild/ProactiveFatBuildSync.java]: Why we need `AtomicInteger` here? May be List should be concurrent too?

File: ignite-tc-helper-web/src/main/java/org/apache/ignite/ci/teamcity/ignited/fatbuild/ProactiveFatBuildSync.java [hunk 3]
```diff
@@ -239,16 +256,32 @@ public void invokeLaterFindMissingByBuildRef(String srvName, ITeamcityConn conn)
...
         }
 
-        return reloadBuild(conn, buildId, existingBuild);
+        FatBuildCompacted savedVer = reloadBuild(conn, buildId, existingBuild);
+
+        BuildRefCompacted refCompacted = new BuildRefCompacted(savedVer);
+        if (savedVer.isFakeStub())
+            refCompacted.setId(buildId); //to provide possiblity to save the build
+
+        final String srvNme = conn.serverId();
```
Comment (SomeFire) [inline ignite-tc-helper-web/src/main/java/org/apache/ignite/ci/teamcity/ignited/fatbuild/ProactiveFatBuildSync.java]: `srvNme` -> `srvName`

File: ignite-tc-helper-web/src/main/java/org/apache/ignite/ci/teamcity/ignited/fatbuild/ProactiveFatBuildSync.java (hunk 2 above)
Comment (dspavlov) [reply ignite-tc-helper-web/src/main/java/org/apache/ignite/ci/teamcity/ignited/fatbuild/ProactiveFatBuildSync.java]: We don't need a volatile semantic as the stream is not parallel, but we used AtomicInt as a counter. I know that it is a kind of antipattern according to JCIP by Brian Goetz, but I find it more readable than creating of an array int[1] and incrementing the value of v[0]+=value. So I suggest to keep it as is for now and probably create later some common non-volatile counter.

**Pull request closed without merging**